Documentation/src/inspect.md

```diff
@@ -0,0 +1,36 @@
+# `IEnumerable` extension method `.Inspect`
+With the `.Inspect` extension method, you can invoke an action for each item in an enumerable, 
+just like `.ForEach` or the `for each` statement would allow you to do, but the method yields the initial enumerable back.

```

Comment:
This description doesn't address when the side effects are executed.

Demonstration:
```csharp
Enumerable.Range(1, 100)
  .Inspect(n => Console.WriteLine($"before where: {n}")
  .Where(n => n % 2 == 0)
  .Inspect(n => Console.WriteLine($"after where: {n}")
  .Inspect(Console.WriteLine)
  .Take(2)
  .ToImmutableList(); // <- Side effects happen here
// Prints:
// before where: 1
// before where: 2
// after where: 2
// before where: 3
// before where: 4
// after where: 4
```

Reply:
Resolved in 0d0bb29